challenge_8/C/randomDeepCopy.c

```diff
@@ -0,0 +1,111 @@
+/*
+Author: karanchawla
+Random Deep Copy
+#Challenge #8
+*/
+
+#include <stdio.h> 
+#include <stdlib.h> 
+
+
+//node definition for the linked list
+struct node
+{
+	int data;
+	struct node* next;
+	struct node* random;
+};
+
+typedef struct node* Node;
+
+//utility function to create a node
+Node newNode(int key)
+{
+	Node temp = (struct node*) malloc(sizeof(struct node));
+
+	temp->data = key;
+	temp->next = NULL;
+	temp->random = NULL;
+
+	return temp;
+}
+
+//utility function to print the list
+void printList(Node head)
```

Comment:
This should also be `Node*`, not `Node`, since you're accessing it as a pointer rather than a literal in the function.